x-pack/plugins/security/server/authentication/authentication_service.ts

```diff
@@ -162,19 +127,44 @@ export class AuthenticationService {
     });
 
     this.logger.debug('Successfully registered core authentication handler.');
-
-    return {
-      getCurrentUser,
-    };
   }
 
-  start({ clusterClient, http }: AuthenticationServiceStartParams): AuthenticationServiceStart {
+  start({
+    audit,
+    config,
+    clusterClient,
+    featureUsageService,
+    http,
+    legacyAuditLogger,
+    loggers,
+    session,
+  }: AuthenticationServiceStartParams): AuthenticationServiceStart {
     const apiKeys = new APIKeys({
       clusterClient,
       logger: this.logger.get('api-key'),
       license: this.license,
     });
 
+    const getCurrentUser = (request: KibanaRequest) => {
+      if (!this.license.isEnabled()) {
```

Comment:
**question** I know you just moved this as-is from `setup`, but is there a reason we need to perform a license check here?

If the license doesn't enable security, then we'd presumably not have an `AuthenticatedUser` on the request state, so we'd still end up returning `null`.

Reply:
Good point, let me double check.